src/jsxc.lib.xmpp.mam.js

```diff
@@ -27,14 +27,111 @@
  *
  */
 
-/*jsxc.xmpp.mam = {
+jsxc.xmpp.mam = {
   conn: jsxc.xmpp.conn
 };
 
 jsxc.xmpp.mam.init = function(){
-  var self = jsxc.xmpp.mam;
-  self.getHistory(userid, count, before);
-};*/
+  var self = jsxc.xmpp.conn.mam;
+  console.log(self);
+
+  //self has now the object of the strophe.mam
+  // make a call to the get history function.
+
+
+  //self.getHistory(userid, count, before);
+};
+
+
+jsxc.xmpp.mam.getHistory = function(bid) {
+  /*
+  To include the whole functionality of strophe-plugin we used self
+  */
+  var self = jsxc.xmpp.conn.mam;
```

Comment:
You have to make sure, that the connection has already be established. See e.g. [jsxc.lib.xmpp.chatState.js#L17](https://github.com/jsxc/jsxc/blob/master/src/jsxc.lib.xmpp.chatState.js#L17)